client/src/setupProxy.js

```diff
@@ -0,0 +1,17 @@
+const { createProxyMiddleware } = require("http-proxy-middleware");
+
+console.log("Setting up a Proxy to localhost:5000");
+module.exports = function (app) {
+  const proxy = createProxyMiddleware({
+    target: "http://localhost:5000",
+    changeOrigin: true,
+  });
+  app.use("/api", proxy);
+  app.use("/_+(flaws|translations|open|document)", proxy);
+  // E.g. search-index.json or index.json
+  app.use("**/*.json", proxy);
+  // This has to match what we do in server/index.js in the catchall handler
+  app.use("**/*.(png|webp|gif|jpe?g|svg)", proxy);
+  // All those root-level images like /favicon-48x48.png
+  app.use("/*.(png|webp|gif|jpe?g|svg)", proxy);
+};
```

Comment:
The default proxy you get with `create-react-app` works such that any file lookup that isn't found on disk is proxied to whatever you have set as `proxy` in your `package.json`. 
That has the problem that URLs that get caught in the react-router stuff in `app.tsx` gets treated as a react-router path. Basically, it will default to serving the `/index.html` plus the client-side react-router stuff. 

To be able to reach other files, setting up a explicit and deliberate router like this is the old way I know. 
Now, any URL that comes in to `localhost:3000` will first go through this list of regexes. Anything that isn't matched will default to `/index.html` which is that page that has the JS bundle with react-router in it.